### ISSUE TITLE
Black screens and Xserver failures

### ISSUE THREAD

**Issue by soulesb** [dnschneid/crouton#3078]
name: jess
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/jess...
crouton: version 1-20170214182546~master:163e0d26
release: jessie
architecture: amd64
xmethod: xiwi
targets: xfce,xiwi,extension
host: version 9000.82.0 (Official Build) stable-channel enguarde 
kernel: Linux localhost 3.10.18 #1 SMP Wed Feb 1 15:02:14 PST 2017 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/jess...
name: precise
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/precise...
crouton: version 1-20170214182546~master:163e0d26
release: precise
architecture: amd64
xmethod: xorg
targets: xfce
host: version 9000.82.0 (Official Build) stable-channel enguarde 
kernel: Linux localhost 3.10.18 #1 SMP Wed Feb 1 15:02:14 PST 2017 x86_64 x86_64 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/precise...
name: swen
encrypted: no
Entering /mnt/stateful_partition/crouton/chroots/swen...
crouton: version 1-20170214182546~master:163e0d26
release: jessie
architecture: amd64
xmethod: xorg
targets: xfce,x11,core,extension,keyboard
host: version 9000.82.0 (Official Build) stable-channel enguarde 
kernel: Linux localhost 3.10.18 #1 SMP Wed Feb 1 15:02:14 PST 2017 x86_64 GNU/Linux
freon: yes
Unmounting /mnt/stateful_partition/crouton/chroots/swen...

So, my issues began with the reported black screen problems for my ubuntu chroots, when I ran sudo startxfce4 the screen would go black and I would have to exit the chroot window with the normal key command of ctrl+alt+shift+back. After trying this with all of my ubuntu chroots, updating them and going to the same state, I deleted the offending chroots. At that point to reinstall new versions of the ubuntu chroots, not thinking to update crouton. These new chroots proceeded to black screen my computer when I ran sudo startxfce4.
At this point I gave up on trying to use an ubuntu chroot, because my old debian chroot was booting properly. I tried installing a chroot with the options -t jessie -u xfce,x11,core,extension,keyboard which promptly black screened. So, I looked around to see if I was doing anything out of order and I thought that I hadn't updated crouton in a while... So, I updated crouton, then without thinking I updated my old Jessie chroot. This caused the chroot to be unable to boot with the following error message:
Cannot connect to extension, retrying...
/usr/local/bin/croutoncycle: 503: kill: No such process

Cannot connect to extension, retrying...
Unable to start display, make sure the crouton extension is installed
and enabled, and up to date. (download from http://goo.gl/OVQOEt)
Running exit commands...
/usr/bin/xinit: connection to X server lost

waiting for X server to shut down Hangup
Hangup 
When I try to boot into my precise root, I get the following error message
xfdesktop[8404]: starting up

(xfce4-session:8392): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(xfwm4:8398): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(Thunar:8402): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(xfsettingsd:8422): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
xfce4-settings-helper: Another instance is already running. Leaving...

(polkit-gnome-authentication-agent-1:8408): GLib-CRITICAL **: g_variant_new_string: assertion `string != NULL' failed

(polkit-gnome-authentication-agent-1:8408): polkit-gnome-1-WARNING **: Failed to register client: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.gnome.SessionManager was not provided by any .service files

(xfwm4:8398): GLib-CRITICAL **: g_str_has_prefix: assertion `prefix != NULL' failed

(migrate:8421): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(xfce4-settings-helper:8426): GLib-CRITICAL **: g_str_has_prefix: assertion `prefix != NULL' failed

(xfce4-settings-helper:8426): xfce4-settings-helper-WARNING **: Failed to get the _NET_NUMBER_OF_DESKTOPS property.

(xfce4-settings-helper:8426): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",

(xfwm4:8398): xfwm4-WARNING **: The property '/general/double_click_distance' of type int is not supported

(xfdesktop:8404): Gtk-WARNING **: Unable to locate theme engine in module_path: "pixmap",
(II) AIGLX: Suspending AIGLX clients for VT switch
^C/usr/bin/xinit: connection to X server lost

Now this is getting to be a long post, but in order to replicate this issue use the normal chroot installation command sudo sh ~/Downloads/crouton -t xfce, with the latest version of crouton: crouton: version 1-20170214182546~master:163e0d26, the latest update of chrome OS. Then run sudo startxfce4. If it is relevant I am using a Lenovo N21 chromebook.

**Comment by DaleTheGoat** [comment]
I'm having the same problem, I believe it has to do with the new ChromeOS update.

**Comment by psiminelakis** [comment]
I have the same issue with the black screen. However, I am able to use `sudo startxfce4 -X xiwi-tab` as suggested in #2970.

**Comment by ahidalgob** [comment]
@psiminelakis thanks for the workaround meanwhile!

**Comment by soulesb** [comment]
I think that the black screens are related to the xmethod xorg. I was abe to connect to my chroot jess using sudo startxfce4, however when I tried to use sudo startxfce4 on swen it launched to a black screen. Once I updated swen with the xiwi target, the chroot booted into a graphical environment normally.
The same fix that I used for swen, installing the xiwi target with sudo sh crouton -u -n precise -t xiwi, allowed precise to launch a graphical environment instead of a black screen.
